pkg/apis/core/validation/validation.go

```diff
@@ -6914,6 +6915,32 @@ func ValidateNodeSpecificAnnotations(annotations map[string]string, fldPath *fie
 	return allErrs
 }
 
+func validateNodeDeclaredFeatures(nodeFeatures []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.NodeDeclaredFeatures) {
+		if len(nodeFeatures) > 0 {
+			allErrs = append(allErrs, field.Forbidden(fldPath, "field is not enabled, feature gate NodeDeclaredFeatures is not set"))
+		}
+		return allErrs
+	}
+
+	featureCount := len(nodeFeatures)
+	for i, feature := range nodeFeatures {
+		if err := ndffeatures.ValidateFeatureName(feature); err != nil {
```

Comment:
I think we should move this function into this package (core/validation). Otherwise, the `ndffeatures` package will require API approval.